src/main/java/cn/nukkit/network/protocol/StartGamePacket.java

```diff
@@ -224,7 +224,12 @@ public void encode() {
         this.putBoolean(false); // Nether type
         this.putString(""); // EduSharedUriResource buttonName
         this.putString(""); // EduSharedUriResource linkUri
-        this.putBoolean(Server.getInstance().isEnableExperimentMode()); // force Experimental Gameplay
+        if (Server.getInstance().isEnableExperimentMode()) { // force Experimental Gameplay
+            this.putBoolean(Server.getInstance().isEnableExperimentMode()); // force Experimental Gameplay
+            this.putBoolean(!Server.getInstance().getConfig("settings.waterdogpe", false)); // Why WaterDogPE require an extra optional boolean if this is set to true? I don't know.
```

Comment:
for people who don't use WaterdogPe or vice versa, a way for the server to know whether or not waterdogpe is active